core/notify.py

```diff
@@ -0,0 +1,11 @@
+from plyer import notification
+
+class Notify:
+    
+    def doNotify(self, title, message, app_name, app_icon):
+        notification.notify(
+            title=title,
+            message=message,
+            app_name=app_name,
+            app_icon=app_icon
+        )
```

Comment:
Por que precisa do `self` aqui?

Reply:
A linguagem python não foi implementada para receber a instancia da própria classe de forma automática. Nesse caso o self é usado para representar a instancia dessa classe. Só assim eu consigo eu consigo ter acessar à métodos da própria classe, sem ter que ficar instanciando ela. Sem falar que não compila quando não tem esse cara. Em uma pesquisa rápida, também vi que o python não faz referencias com `@`.

Reply:
Apenas como comentário, mas essa pergunta é muito interessante, porque é uma peculiaridade da linguagem, e durante a codificação isso acaba trazendo perguntas do tipo "porque essa m.. não funciona ?"

Reply:
Bem interessante essa informação, não fazia ideia disso. No meu ver parece ser uma limitação, você ter que informar a própria instancia. Você que trabalha mais tempo com Python, isso tem algum sentido ou uso posterior para precisar desse jeito?

Reply:
Para usar posteriormente, não, tanto é que na invocação da classe você simplesmente não informa esse parâmetro. Utilizo o `self` a penas para referenciar a classe dentro de um método nela mesma.
Quanto a parecer limitação, acho difícil. Deve ter algum motivo pra ser assim. Se pá alguma coisa relacionada a recursos, porque ele envia a instancia, mas não recebe automaticamente. (vamos ter que dar uma pesquisada nisso :dagger: )
Um fato curioso nesse caso, é que `self` não é uma palavra reservada da linguagem

Reply:
achei uma para interessante:
 - **"Explicit is better than implicit"**

Essa frase está nos "princípios" do[ **The Zen of Python**](https://www.python.org/dev/peps/pep-0020/)

Mas pelo visto teve mais gente com essa preocupação: 

[@staticmethod](https://www.programiz.com/python-programming/methods/built-in/staticmethod)

```
Self Can Be Avoided
By now you are clear that the object (instance) itself is passed along as the first argument, automatically. This implicit behavior can be avoided while making a static method. Consider the following simple example:

class A(object):

    @staticmethod
    def stat_meth():
        print("Look no self was passed")
```